### PR TITLE
Docker upgrade database

### PR DESCRIPTION
**Pull request by dhaley**
#### Any background context you want to provide?

The script `./docker/postgres_upgrade.sh` will properly upgrade a database from pg12.7-(2.3.0) to pg14.11-(2.14.2).  It works in conjunction with the "docker-compose.pgupgrade.yml" docker-compose file to temporarily run Timescale Postgres 13 and 14 images.

#### How should this be manually tested?

./docker/postgres_upgrade.sh  --source-pg-conv-version 13 --source-ts-conv-version 2.11.2 --target-pg-test-version 14 --target-ts-test-version 2.11.0

### CODE REVIEW

File: docker/postgres_upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+docker-compose exec -T seed-pg13 psql --user=seed -c 'GRANT ALL PRIVILEGES ON DATABASE "seed" TO "seed";'
+
+# Alter the user "seed" to have CREATEDB, CREATEROLE, and SUPERUSER privileges
+docker-compose exec -T seed-pg13 psql --user=seed -c 'ALTER USER "seed" CREATEDB CREATEROLE SUPERUSER;'
+
+# Create the PostGIS extension if it does not exist
+docker-compose exec -T seed-pg13 psql --user=seed -d seed -c 'CREATE EXTENSION IF NOT EXISTS postgis;'
+
+# Create the TimescaleDB extension if it does not exist
+docker-compose exec -T seed-pg13 psql --user=seed -d seed -c "CREATE EXTENSION IF NOT EXISTS timescaledb WITH VERSION '2.3.0';"
```
Comment (axelstudios) [inline docker/postgres_upgrade.sh:67]: This won't do anything without dropping the extension first:
`NOTICE:  extension "timescaledb" already exists, skipping`

File: docker/postgres_upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+docker-compose exec -T seed-pg14 psql --user=seed -d seed -c 'SELECT timescaledb_pre_restore();'
+
+echo "Restoring postgres 13 database ${PG13_DUMP} from backup..."
+# Restore from backup
+# Temporarily disable error stopping
+set +e
+docker-compose exec -T seed-pg14 pg_restore --no-owner --no-acl --user=seed -d seed -v ${PG13_DUMP}
+# Re-enable error stopping
+set -e
+
```
Comment (axelstudios) [inline docker/postgres_upgrade.sh:150]: Missing the post_restore step

**Review by axelstudios — CHANGES_REQUESTED** [review]

File: docker/postgres_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+echo "Waiting for PostgreSQL to start..."
+# Wait until PostgreSQL starts and is ready to accept connections.
+until docker-compose exec -T db-postgres pg_isready -U seed -d seed > /dev/null 2>&1; do
+    echo -n "."
+    sleep 1
+done
+
+POSTGRES_FULL_VERSION=$(docker-compose exec -T db-postgres psql --user=seed -d postgres -c 'SELECT version();' | sed -n 3p)
+# Extract the version number using regex
+POSTGRES_VERSION=$(echo "$POSTGRES_FULL_VERSION" | sed -n 's/.*PostgreSQL \([0-9]*\.[0-9]*\).*/\1/p')
```
[the remark this answers is not in the document]
Comment (dhaley) [reply docker/postgres_upgrade.sh:17]: Thanks. Will change.

**Comment by axelstudios** [comment]
Might also be worth using the `--exit-on-error` flag with pg_restore

File: docker/postgres_upgrade.sh (hunk 1 above)
Comment (dhaley) [reply docker/postgres_upgrade.sh:67]: Good catch.

**Comment by dhaley** [comment]
> Might also be worth using the `--exit-on-error` flag with pg_restore

Will do.

**Review by axelstudios — APPROVED** [review]

**Review by nllong — APPROVED** [review]
This is great, thanks @dhaley!

**Pull request merged**